main.go

```diff
@@ -188,6 +190,31 @@ func updateWatchdogTimer(a *attiny) {
 	}
 }
 
+func updateWifiState(a *attiny) {
+	connectedState := false
+	for {
+		if connectedState != wifiConnected() {
+			if err := a.SetWifiState(!connectedState); err != nil {
+				log.Println("error with updating wifi state")
+				time.Sleep(time.Minute)
+			} else {
+				connectedState = !connectedState
+				log.Printf("updated wifi connected state to '%t'", connectedState)
+			}
+		}
+		time.Sleep(time.Second)
```

Comment:
Running this command every second is quite inefficient. The best way to approach this problem is to use dhcpcd's hook scripts so that attiny-controller is notified when network interfaces change. This avoids polling.

Drop a script in `/libexec/dhcpcd-hooks` (call it something like `10-notify-attiny`) which calls a attiny-controller D-Bus API to let it know if WiFi is up or not. A number of environment variables are available in hook scripts - `$interface` is probably the most useful one. The script could just not doing anything unless `$interface` is the wifi interface and then do what `wifiConnected` is doing here and then send the result to attiny-controller.

See:
- https://www.daemon-systems.org/man/dhcpcd.8.html
- https://jlk.fjfi.cvut.cz/arch/manpages/man/dhcpcd-run-hooks.8